src/python/CRABInterface/RESTTask.py

```diff
@@ -76,6 +76,75 @@ def summary(self, **kwargs):
         return rows
 
+    # short status summary for the UI MAIN tab
+    def status(self, **kwargs):
+        """
+         The API is (only?) used in the monitor for operator.
+           curl -X GET 'https://mmascher-dev6.cern.ch/crabserver/dev/task?subresource=statush&workflow=150224_230633:mmascher_crab_testecmmascher-dev6_3' \
+                        -k --key /tmp/x509up_u8440 --cert /tmp/x509up_u8440 -v
+        to be used in the CRABServer UI to display a summary of Task status
+        moved here from RESTWorkflow to have it together with the "search" below
+        after the deprecation of old worfklow/status API which was calling HTCondor inside
+        """
+        # this retuns a JSON "ready to be displayed" of the form {key:value,...}
+        # different from the format of many REST API
+        # {'desc':{'columns':[list of column names]}, 'result':[list of column values]}
+        # wo we need to repeat the SQL query used in old code in
+        # https://github.com/dmwm/CRABServer/blob/5c7777045b33302fb97128ed29d4627141e009ec/src/python/CRABInterface/HTCondorDataWorkflow.py#L170
+        # with the two steps:
+        # 1. retrieve the values
+        # 2. assign them to a name ntuple so to be able to address them
+
+        row = self.api.query(None, None, self.Task.ID_sql, taskname = kwargs['workflow'])  # step 1. get values
+        try:
+            #just one row is picked up by the previous query
+            row = self.Task.ID_tuple(*next(row))  # step 2. assign name to values
+        except StopIteration:
+            raise ExecutionError("Impossible to find task %s in the database." % workflow)
+
+        # now pick code from old implementation
+        # Empty results, only fields for which a non NULL value was retrieved from DB will be filled
+        result = {"submissionTime"   : '',
+                  "username"         : '',
+                  "status"           : '',
+                  "command"          : '',
+                  "taskFailureMsg"   : '',
+                  "taskWarningMsg"   : [],
+                  "splitting"        : '',
+                  "schedd"           : '',
+                  "taskWorker"       : '',
+                  "webdirPath"       : '',
+                  "clusterid"        : ''}
+
+        result['submissionTime'] = str(row.start_time)  # convert from datetime object to readable string
+        result['status'] = row.task_status
+        if row.task_command:
+            result['command'] = row.task_command
+        if row.task_failure is not None:
+            if isinstance(row.task_failure, str):
+                result['taskFailureMsg'] = row.task_failure
+            else:
+                result['taskFailureMsg'] = row.task_failure.read()
+        # Stefano does not know why taskWarningMsg is handled differently from taskFailureMsg, quite possibly
+        # the simplest form above works alwasy in a PY3 world
```

Comment:
I guess it is for backward compatibility for py2, and you need `literal_eval` to parse list of strings from clob column.

Reply:
thanks. I will add an explanation as comment in the code